app/optscale_api/auth_api.py

```diff
@@ -2,95 +2,74 @@
 
 import logging
 
+from fastapi import status as http_status
+
 from app import settings
 from app.core.api_client import APIClient
+from app.core.exceptions import OptScaleAPIResponseError, UserAccessTokenError
 
-AUTH_TOKEN_ENDPOINT = "/auth/v2/token"
+AUTH_TOKEN_ENDPOINT = "/auth/v2/tokens"
 logger = logging.getLogger("optscale_auth_api")
 
 
-class OptScaleAuth:
-    def __init__(self):
-        self.api_client = APIClient(base_url=settings.opt_scale_api_url)
+def build_admin_api_key_header(admin_api_key: str) -> dict[str, str]:
+    """
+    Builds headers with the admin API key for root-level operations
+    :param admin_api_key: the secret API key
+    :type admin_api_key: string
+    :return: a dictionary {"Secret": "secret key here"}
+    """
+    return {"Secret": admin_api_key}
```

Comment:
Aren't these two functions a bit overengineered? I initially got scared when I looked at this file but the authentication is actually much simpler 😆

Reply:
Well, probably. However, I don't see any reason to be scared :)